specification/protocol/generate_rest.yaml

```diff
@@ -0,0 +1,89 @@
+openapi: 3.1.0
+info:
+  title: Open Inference API for text generation
+  description: Open Inference API for text generation
+  version: 1.0.0
+components:
+  schemas:
+    GenerateRequest:
+      type: object
+      required: 
+        - text_input
+      properties:
+        text_input:
+          type: string
+        parameters:
+          $ref: '#/components/schemas/GenerateParameters'
+    GenerateParameters:
+      type: object
+      properties:
+        parameter:
+          description: Parameters are framework specific. An optional object containing zero or more parameters for this generate request expressed as key/value pairs 
+    GenerateResponse:
+      type: object
+      required:
+        - text_output
+        - model_name
+        - model_version
+      properties:
+        text_output:
+          type: string
+        model_name:
+          type: string
+        model_version:
+          type: string
+    GenerateStreamResponse:
+      type: object
+      required:
+        - text_output
+        - model_name
+        - model_version
+        - done
+      properties:
+        text_output:
+          type: string
+        model_name:
+          type: string
+        model_version:
+          type: string
+paths:
+  /v2/models/${MODEL_NAME}[/versions/${MODEL_VERSION}]/generate:
+    post:
+      parameters:
+        - name: model_name
+          required: true
+          in: path
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/GenerateRequest'
+      responses:
+        '200':
```

Comment:
Can we also define the error responses ?

Reply:
I think we would want to define a list of http error response with the `GenerateErrorResponse` like here
https://github.com/huggingface/text-generation-inference/blob/main/docs/openapi.json#L50

Reply:
Thanks! will get this updated